jobserver/releases.py

```diff
@@ -60,29 +58,35 @@ def handle_release(workspace, backend, backend_user, release_hash, upload):
             backend=backend,
             backend_user=backend_user,
             upload_dir=upload_dir,
-            files=[str(f) for f in files],
+            files=files,
         )
 
     except Exception:
         # something went wrong, clean up files, they will need to reupload
-        shutil.rmtree(actual_dir, ignore_errors=True)
+        if actual_dir:
+            shutil.rmtree(actual_dir, ignore_errors=True)
         raise
 
     return release, True
 
 
-def list_files(directory):
-    """Utility to list files in a directory tree."""
-    return list(
-        sorted(p.relative_to(directory) for p in directory.glob("**/*") if p.is_file())
-    )
-
-
-def hash_files(directory, files):
+def hash_files(directory):
     # use md5 because its fast, and we only care about uniqueness, not security
+    files = list(
+        sorted(
+            str(p.relative_to(directory)) for p in directory.glob("**/*") if p.is_file()
+        )
+    )
```

Comment:
Why do you generate the file list inside this function instead of passing it in?  (and can you add a comment to explain please 🙂)

Reply:
hmm, just did it to avoid a needless one-line function that was only ever used in one place.

Reply:
Makes sense 👍